### PR TITLE
Use archives instead of debs

### LINKED ISSUE #30: Use of archives instead of deb packages
Dear developers,

wouldn't it be easier to use the mendeley packages instead of the debian packages? The corresponding code would look like this

```
    sources:
    - only-arches: [i386]
      type: archive
      url: https://desktop-download.mendeley.com/download/linux/mendeleydesktop-1.19.4-linux-i486.tar.bz2
      sha256: 00e8047d1c2cb71db7ded001841e95d37415c72884403bdcb24bb99895f090d3
      x-checker-data:
	type: rotating-url
	url: https://www.mendeley.com/autoupdates/installer/Linux-x86/stable-incoming
    - only-arches: [x86_64]
      type: archive
      url: https://desktop-download.mendeley.com/download/linux/mendeleydesktop-1.19.4-linux-x86_64.tar.bz2
      sha256: 9700f2db273bb364aadb92535ba40349626193e0ba5ff8544ce22b665d5d7dd9
      x-checker-data:
	type: rotating-url
	url: https://www.mendeley.com/autoupdates/installer/Linux-x64/stable-incoming
```

### PR DESCRIPTION
**Pull request by jakobjakobson13**
Would fix #30

### CODE REVIEW

File: com.elsevier.MendeleyDesktop.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - install -Dm644 share/icons/hicolor/22x22/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/22x22/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/32x32/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/32x32/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/48x48/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/48x48/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/64x64/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/128x128/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png
+      - mkdir -p /app/share/mendeleydesktop
+      - cp -r share/mendeleydesktop/* /app/share/mendeleydesktop
+    sources:
+    - type: archive
+      only-arches: [i386]
```
Comment (barthalion) [inline com.elsevier.MendeleyDesktop.yml:37]: This can be deleted as well.

File: com.elsevier.MendeleyDesktop.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      - install -Dm644 share/icons/hicolor/16x16/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/16x16/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/22x22/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/22x22/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/32x32/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/32x32/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/48x48/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/48x48/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/64x64/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png
+      - install -Dm644 share/icons/hicolor/128x128/apps/mendeleydesktop.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/${FLATPAK_ID}.png
+      - mkdir -p /app/share/mendeleydesktop
+      - cp -r share/mendeleydesktop/* /app/share/mendeleydesktop
+    sources:
+    - type: archive
```
Comment (barthalion) [inline com.elsevier.MendeleyDesktop.yml:36]: It has to be an extra-data app. We don't have redistribution rights.

File: com.elsevier.MendeleyDesktop.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  - --filesystem=/tmp
+modules:
+  - name: mendeley
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - install bin/mendeleydesktop /app/bin
+      - mkdir -p /app/lib
+      - cp -r lib/* /app/lib
+      - install -Dm644 share/applications/mendeleydesktop.desktop /app/share/appdata/com.elsevier.MendeleyDesktop.appdata.xml
```
Comment (barthalion) [inline com.elsevier.MendeleyDesktop.yml:26]: You install desktop file as appdata.

**Review by barthalion — CHANGES_REQUESTED** [review]

**Comment by Erick555** [comment]
Thx for your contribution, however switch for using deb instead of bz2 [was a conscious choice which saves bandwidth every time user downloads/updates app](https://github.com/flathub/com.elsevier.MendeleyDesktop/pull/16). Also i386 arch is no more on flathub with 19.08 runtime.

**Pull request closed without merging**